lib/i18n/js/engine.rb

```diff
@@ -9,56 +10,58 @@ class SprocketsExtension
-        when lambda {|v| v2_only.match?("", v) || v3_plus.match?("", v) }
-          { after: :engines_blank_point, before: :finisher_hook }
-        else
-          raise StandardError, "Sprockets version #{sprockets_version} is not supported"
-        end
+      # `sprockets.environment` was used for 1.x of `sprockets-rails`
+      # https://github.com/rails/sprockets-rails/issues/227
+      #
+      # References for current values:
+      #
+      # Here is where sprockets are attached with Rails. There is no 'sprockets.environment' mentioned.
+      # https://github.com/rails/sprockets-rails/blob/master/lib/sprockets/railtie.rb
+      #
+      # Finisher hook is the place which should be used as border.
+      # http://guides.rubyonrails.org/configuring.html#initializers
+      #
+      # For detail see Pull Request:
+      # https://github.com/fnando/i18n-js/pull/371
+      #
+      # Not using -> for JRuby compatibility
+      # See https://github.com/fnando/i18n-js/issues/419
+      initializer_args = case sprockets_version
+      when lambda {|v| v2_only.match?("", v) || v3_plus.match?("", v) }
+        { after: :engines_blank_point, before: :finisher_hook }
+      else
+        raise StandardError, "Sprockets version #{sprockets_version} is not supported"
+      end
 
-        initializer "i18n-js.register_preprocessor", initializer_args do
-          # This must be called inside initializer block
-          # For details see comments for `using_asset_pipeline?`
-          next unless JS::Dependencies.using_asset_pipeline?
+      initializer "i18n-js.register_preprocessor", initializer_args do
+        # This must be called inside initializer block
+        # For details see comments for `using_asset_pipeline?`
+        next unless JS::Dependencies.using_asset_pipeline?
 
-          # From README of 2.x & 3.x of `sprockets-rails`
-          # It seems the `configure` block is preferred way to call `register_preprocessor`
-          # Not sure if this will break older versions of rails
-          #
-          # https://github.com/rails/sprockets-rails/blob/v2.3.3/README.md
-          # https://github.com/rails/sprockets-rails/blob/v3.0.0/README.md
+        # From README of 2.x & 3.x of `sprockets-rails`
+        # It seems the `configure` block is preferred way to call `register_preprocessor`
+        # Not sure if this will break older versions of rails
+        #
+        # https://github.com/rails/sprockets-rails/blob/v2.3.3/README.md
+        # https://github.com/rails/sprockets-rails/blob/v3.0.0/README.md
+        if JS::Dependencies.sprockets_supports_register_preprocessor?
           Rails.application.config.assets.configure do |config|
             config.register_preprocessor(
               "application/javascript",
               ::I18n::JS::SprocketsExtension,
             )
           end
+        else
+          Rails.application.assets.register_preprocessor "application/javascript", ::I18n::JS::SprocketsExtension
```

Comment:
Yeah, latest version of sprockets supported by rails 3 is 2.2 and it doesn't expose static register_preprocessor method to Sprockets class. 

`Rails.application.config.assets` is an instance of `Sprockets::Railtie::OrderedOptions` in Rails 4, while it's simply `ActiveSupport::OrderedOptions` in rails 3, so calling configure does nothing for rails 3